src/Toml/Codec/Combinator/Custom.hs

```diff
@@ -235,3 +241,18 @@ validateIf p biMap k = validate validateEither biMap k
         if p a
         then Right a
         else Left $ "Value does not pass the validation for key: " <> prettyKey k
+
+{- | Converts a given @TomlCodec@ to give you access to the list of
+@TomlDecodeError@s as an @Either@
+
+__Example:__
+
+@
+either (bool "mykey") :: TomlCodec (Either [TomlDecodeError] Bool)
+@
+-}
+either :: TomlCodec a -> TomlCodec (Either [TomlDecodeError] a)
+either (Codec r w) = Codec r' $ \case
+  Left x' -> pure . Left $ x'
+  Right x' -> fmap Right . w $ x'
+  where r' = pure . validationToEither . r
```

Comment:
Let's rename variables for clarity, and also use `Success` here for clarity. Just to make the code more maintainable :slightly_smiling_face: 
```suggestion
  where
    newCodecRead :: TomlEnv (Either [TomlDecodeError] a)
    newCodecRead = Success . validationToEither . oldCodecRead
```

Reply:
Yes I agree, I moved out both `newCodecRead` and `newCodecWrite`